modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java

```diff
@@ -405,6 +405,10 @@ public String toEnumName(CodegenProperty property) {
                 if (Boolean.TRUE.equals(var.isEnum)) {
                     var.datatypeWithEnum = var.datatypeWithEnum.replace(var.enumName, cm.classname + "." + var.enumName);
                 }
+
+                if (isReservedWord(var.name)) {
```

Comment:
@JFCote Shall we do it in `toVarName` instead similar to other generators? e.g. https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/RubyClientCodegen.java#L453

Reply:
@wing328 As you can see here: https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractTypeScriptClientCodegen.java#L169

In typescript, the toVarName is simply returning the variable name itself without transformation. Should I change it to the same thing as javascript? 
https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavascriptClientCodegen.java#L490

Let me know and I'll do the changes!

Reply:
Please follow the JS approach, which is used by other generators (e.g. Ruby, PHP, etc)

Reply:
Ok! Will try to commit a fix this week! Thanks for the review!